### PR TITLE
[lldb][NFC] Implement RTTI for DWARFASTParserSwift

### PR DESCRIPTION
**Pull request by augusto2112**

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci smoke test

File: lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp [hunk 1]
```diff
@@ -61,7 +61,8 @@ using namespace lldb;
 using namespace lldb_private;
 using namespace lldb_private::dwarf;
 DWARFASTParserClang::DWARFASTParserClang(TypeSystemClang &ast)
-    : m_ast(ast), m_die_to_decl_ctx(), m_decl_ctx_to_die() {}
+    : DWARFASTParser(Kind::DWARFASTParserClang), m_ast(ast),
+      m_die_to_decl_ctx(), m_decl_ctx_to_die() {}
```
Comment (adrian-prantl) [inline lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:65]: This part of the commit should be made upstream.

Comment (adrian-prantl) [inline lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:65]: Also the () initializers are redundant.

Comment (augusto2112) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:65]: The first two commits are cherry picks from upstream

Comment (adrian-prantl) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:65]: oh!

Comment (adrian-prantl) [reply lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:65]: LGTM!

**Pull request merged**